lib/iris/coords.py

```diff
@@ -143,6 +143,147 @@ def __new__(cls, name_or_coord, minimum, maximum,
                                       'groupby_point, groupby_slice')
 
 
+def discontiguity_check_2d(coord, data, abs_tol=1e-4, transpose=False):
+    """
+    Check bounds of a 2-dimensional coordinate are contiguous, and check that
+    any discontiguous bounds occur where the data is masked.
+
+    If discontiguity occurs but data is masked, raise warning
+    If discontiguity occurs and data is NOT masked, raise error
+
+    Args:
+        coord:
+            Array of the bounds of a 2D coord, of shape (X,Y,4)
+        data:
+            data of the the cube we are plotting
+        abs_tol:
+            tolerance when checking the contiguity
+
+    """
+    # TODO Work out how to tell which direction to check contiguity in:
+    # The bounds that are supplied are the bounds for one axis only, so a
+    # contiguity check will fail if checked along the other axis
+    # i.e.:
+    # latitude_bds = [0, 1, 1, 0]
+    # longitude_bds = [0, 0, 1, 1]
+    # You can see that it matters which axis we are checking as the indexes
+    # we are matching up differ.
+    #
+    # To explain that a bit better, if I pass in the longitude coordinate and
+    # therefore the bounds array is x-values only, then checking that the
+    # x-values match up in the y-direction seems a bit pointless.  It makes
+    # more sense to check that the x-values match up in the x-coord direction
+    # (e.g. longitude) and the y-values match up in the y-coord direction
+    # (e.g. latitude).
+    if transpose:
+        bds = coord.bounds.T
+    else:
+        bds = coord.bounds
+    axis = guess_coord_axis(coord)
+
+    # TODO Rewrite this comment? The bounds indexes should be anticlockwise from bottom-left, not clockwise from top-right.
+    # Check ordering:
+    #         i    i+1
+    #    j    @0    @1
+    #  j+1    @3    @2
+    def mod360_diff(x1, x2):
+        diff = x1 - x2
```

Comment:
Here again, the code is assuming the values must be angles in degrees. 
Can we guarantee that is the only thing that this deals with,
and if not can we either distinguish this case or generalise the operation ?